common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/IDToken.java

```diff
@@ -219,27 +217,27 @@ public String getRawIDToken() {
         return Collections.unmodifiableMap(mTokenClaims);
     }
 
-    private Map<String, String> parseJWT(final String idToken) throws ServiceException {
-        final String idTokenBody = extractJWTBody(idToken);
-        final byte[] data = Base64.decode(idTokenBody, Base64.URL_SAFE);
-
+    public static Map<String, String> parseJWT(final String idToken) throws ServiceException {
         try {
-            final String decodedBody = new String(data, Charset.forName(StringExtensions.ENCODING_UTF8));
-            return JsonExtensions.extractJsonObjectIntoMap(decodedBody);
-        } catch (final JSONException e) {
-            throw new ServiceException("", ErrorStrings.INVALID_JWT, e);
-        }
-    }
+            // Create a SignedJWT from the input token String
+            final SignedJWT signedJWT = SignedJWT.parse(idToken);
+
+            // Grab the claims and put them in a Map
+            final JWTClaimsSet claimsSet = signedJWT.getJWTClaimsSet();
+            final Map<String, Object> claimsMap = claimsSet.getClaims();
 
-    private String extractJWTBody(final String idToken) {
-        final int firstDot = idToken.indexOf('.');
-        final int secondDot = idToken.indexOf('.', firstDot + 1);
-        final int invalidDot = idToken.indexOf('.', secondDot + 1);
+            // Convert that Map<String, Object> into Map<String, String>
+            final Map<String, String> claimsMapStr = new HashMap<>();
 
-        if (invalidDot == -1 && firstDot > 0 && secondDot > 0) {
-            return idToken.substring(firstDot + 1, secondDot);
-        } else {
-            throw new IllegalArgumentException("Invalid ID token format.");
+            for (final Map.Entry<String, Object> entry : claimsMap.entrySet()) {
+                claimsMapStr.put(entry.getKey(), entry.getValue().toString());
+            }
+
+            // Return our result
+            return claimsMapStr;
+        } catch (ParseException e) {
+            throw new ServiceException("Failed to parse JWT", ErrorStrings.INVALID_JWT, null);
```

Comment:
Can we move this catch block to line 227 and pass the ParseException to ServiceException